### PR TITLE
fix(url-parser): merge browser and node url parser, add rn url parser

### LINKED ISSUE #1867: "Error: not implemented" in React Native environment
**Describe the bug**
Aws SDK for JavaScript v3 throws "Error: not implemented" when run in React Native environment.

**SDK version number**
`3.1.0`

**Is the issue in the browser/Node.js/ReactNative?**
React Native

**Details of the browser/Node.js/ReactNative version**
`0.63.4`

**To Reproduce (observed behavior)**
Run React Native set up from https://github.com/aws-samples/aws-sdk-js-tests#react-native
The error is thrown in Emulator output.

**Expected behavior**
The output is displayed under JS SDK v3

**Screenshots**

<details>
<summary>Browser</summary>

Screen Shot 2021-01-05 at 10 11 51 AM

</details>

<details>
<summary>React Native</summary>

Screen Shot 2021-01-05 at 9 45 38 AM

</details>

### PR DESCRIPTION
**Pull request by AllanZhengYP**
Fixes #1867, #1900

Both `@aws-sdk/url-parser-node` and `@aws-sdk/url-parser-browser` use WHATWG `URL()`
to parse the url. In fact their code are identical. So they are deprecated and
replaced by `@aws-sdk/url-parser`.

However, `URL()` is not supported in React Native. So we need to use `parse()`
from "url" package. `@aws-sdk/url-parser-native` is used in React Native runtime.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by AllanZhengYP** [comment]
@trivikr  Confirmed that the issue is solved in simulator.
simulator_screenshot_DEB32429-0BF6-4656-A565-85E226DAD309

File: packages/url-parser/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@aws-sdk/url-parser-browser",
```
Comment (trivikr) [inline packages/url-parser/package.json:2]: The values in `homepage` and `repository.directory` also need to be updated.

**Review by trivikr — COMMENTED** [review]
All Changelog below the title `packages/url-parser/CHANGELOG.md` should be removed, as it's a new package.

The package name in `packages/url-parser/README.md` title and npm badges needs to be updated.

File: packages/url-parser/package.json (hunk 1 above)
Comment (AllanZhengYP) [reply packages/url-parser/package.json:2]: Good point. Addressed in https://github.com/aws/aws-sdk-js-v3/pull/1903/commits/b9e068bd5d5102597920dfb751dedeafe40d8926

**Review by trivikr — APPROVED** [review]

**Pull request merged**